### PR TITLE
Expose DB wait/exec times over JMX as well as logs

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
Replaces bespoke log prints with metric collection through the `MetricsManager`, which has both a JMX and an SLF4J sink. Times are now captured on a query-by-query _and_ overall basis.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/JdbcLedgerDao.scala [hunk 1]
```diff
@@ -705,25 +705,23 @@ private class JdbcLedgerDao(
...
-          // We can't use the storeLedgerEntry(), as that one does update the ACS
-          ledgerEntries.foreach {
-            case (i, le) =>
-              le match {
-                case tx: LedgerEntry.Transaction => storeTransaction(i, tx)
-                case rj: LedgerEntry.Rejection => storeRejection(i, rj)
-                case cp: LedgerEntry.Checkpoint => storeCheckpoint(i, cp)
-              }
-          }
+      .executeSql("store_initial_state_from_scenario") { implicit conn =>
```
[the remark this answers is not in the document]
Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/JdbcLedgerDao.scala]: Indeed, suggestions?

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/metrics/MetricsManager.scala [hunk 2]
```diff
@@ -35,6 +35,8 @@ final class MetricsManager(enableJmxReporter: Boolean) extends AutoCloseable {
 
   if (enableJmxReporter) jmxReporter.start()
 
+  def unmanagedTimer(name: String) = metrics.timer(name)
```
[the remark this answers is not in the document]
Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/metrics/MetricsManager.scala]: Addressed by https://github.com/digital-asset/daml/pull/3310/commits/c4516473b10ead76cd7473e13afdc8a83e5b986d

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/JdbcLedgerDao.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/JdbcLedgerDao.scala]: Addressed by https://github.com/digital-asset/daml/pull/3310/commits/c4516473b10ead76cd7473e13afdc8a83e5b986d

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/util/SqlExecutor.scala [hunk 3]
```diff
@@ -30,19 +32,17 @@ class SqlExecutor(noOfThread: Int, loggerFactory: NamedLoggerFactory) extends Au
 
   def runQuery[A](description: => String, block: () => A): Future[A] = {
     val promise = Promise[A]
-    val startWait = System.nanoTime()
+    val waitTimer = mm.unmanagedTimer(s"sql_${description}_wait")
+    val waitAllTimer = mm.unmanagedTimer("sql_all_wait")
```
[the remark this answers is not in the document]
Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/util/SqlExecutor.scala]: Addressed by https://github.com/digital-asset/daml/pull/3310/commits/c4516473b10ead76cd7473e13afdc8a83e5b986d

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/util/DbDispatcher.scala [hunk 4]
```diff
@@ -65,8 +68,9 @@ private class DbDispatcherImpl(
           logger.error(s"got an uncaught exception on thread: ${thread.getName}", t))
         .build()))
 
-  override def executeSql[T](description: => String)(sql: Connection => T): Future[T] =
-    sqlExecutor.runQuery(description, () => connectionProvider.runSQL(conn => sql(conn)))
+  override def executeSql[T](description: String, extraLog: Option[String] = None)(
+      sql: Connection => T): Future[T] =
+    sqlExecutor.runQuery(description)(connectionProvider.runSQL(conn => sql(conn)))
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/util/DbDispatcher.scala]: `extraLog` needs to be passed to `runQuery`.

Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/util/DbDispatcher.scala]: Gah. Thanks!

Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/util/DbDispatcher.scala]: Addresse by https://github.com/digital-asset/daml/pull/3310/commits/da69019b20e9b758d06008c4b2a5ab303bd1e4b6

**Pull request merged**